File: PythonSolutions/ContainsDuplicate_219.py
from typing import List

class Solution:
    def containsNearbyDuplicate(self, nums:List[int],k: int) -> bool:
     window = set()
     for i, num in enumerate(nums):
          if num in window:
            return True
          window.add(num)
          if len(window) > k:
             window.remove(nums[i - k])
     return False


if __name__ == "__main__":
    solution = Solution()
    nums = [1,2,3,1]
    k = 3
    print(solution.containsNearbyDuplicate(nums,k))
    

File: JavaScriptSolutions/ContainsDuplicate_219.js
const containsNearbyDuplicate = (nums, k) => {
  //sliding window
  let window = new Set();

  for (let i = 0; i < nums.length; i++) {
    if (window.has(nums[i])) {
      return true;
    }

    window.add(nums[i]);

    if (window.size > k) {
      window.delete(nums[i - k]);
    }
  }

  return false;
};

const nums = [1, 2, 3, 1];
const k = 3;

console.log(containsNearbyDuplicate(nums, k));
